src/web/src/pages/signup.tsx

```diff
@@ -0,0 +1,271 @@
+import formModels from '../components/SignUp/Schema/FormModel';
+import formSchema from '../components/SignUp/Schema/FormSchema';
+import { usersServiceUrl, webUrl } from '../config';
+
+const {
+  firstName,
+  lastName,
+  displayName,
+  githubUsername,
+  github,
+  githubOwnership,
+  blogUrl,
+  feeds,
+  email,
+  blogOwnership,
+} = formModels;
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      padding: '0',
+      margin: '0',
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      minHeight: '100vh',
+      width: '100vw',
+      boxSizing: 'border-box',
+      position: 'relative',
+      fontSize: '1.1rem',
+    },
+    imageContainer: {
+      minHeight: '100vh',
+      width: '100vw',
+      position: 'absolute',
+      top: '0',
+      bottom: '0',
+      zIndex: -1,
+      [theme.breakpoints.down(600)]: {
+        display: 'none',
+      },
+    },
+    signUpContainer: {
+      margin: '1% 0 1% 0',
+      display: 'grid',
+      gridTemplateRows: '10% auto 15%',
+      gridGap: '2%',
+      justifyItems: 'center',
+      fontFamily: 'Spartan',
+      height: '510px',
+      width: '510px',
+      padding: '1%',
+      borderRadius: '5px',
+      boxShadow: '2px 4px 4px 1px rgba(0, 0, 0, 0.1)',
+      background: '#ECF5FE',
+      '@media (max-height: 500px) and (max-width: 1024px)': {
+        margin: '0 0 65px 0',
+      },
+      [theme.breakpoints.down(600)]: {
+        background: 'none',
+        boxShadow: 'none',
+        minHeight: '650px',
+        height: '600px',
+        position: 'absolute',
+        top: '0px',
+        width: '100%',
+        margin: '0',
+        padding: '0',
+        gridTemplateRows: '8% auto 17%',
+      },
+    },
+    title: {
+      color: '#121D59',
+      fontSize: '22px',
+    },
+    infoContainer: {
+      width: '100%',
+      position: 'relative',
+    },
+    buttonsWrapper: {
+      margin: '0 auto',
+      width: '50%',
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+    },
+    button: {
+      height: '4rem',
+      width: '40%',
+      fontSize: '1.1em',
+      padding: '0.7em',
+      margin: '5px',
+      background: '#E0C05A',
+      '&:hover': {
+        color: 'black',
+        background: '#EBD898',
+      },
+    },
+    buttonLogin: {
+      height: '4rem',
+      width: '40%',
+      fontSize: '1.1em',
+      padding: '0.7em',
+      margin: '5px',
+      background: '#FF0000',
+      color: '#FFF',
+      '&:hover': {
+        background: '#FF7070',
+      },
+    },
+    text: {
+      textAlign: 'center',
+      fontSize: '0.9em',
+      color: '#474747',
+    },
+  })
+);
+
+const SignUpPage = () => {
+  const classes = useStyles();
+  const [activeStep, setActiveStep] = useState<number>(0);
+  const currentSchema = formSchema[activeStep];
+  const { user, token, login } = useAuth();
+  const [loggedIn, setLoggedIn] = useState(false);
```

Comment:
Do we need this vs. using `!!user` below, or `const loggedIn = !!user;` here?

Reply:
We used `!!user` but NextJS gave `Text content did not match` error and the buttons were not updated according to `user` value. This must have something to do with NextJS's server-side-rendering React.
See also https://aboutmonica.com/blog/server-side-rendering-react-hydration-best-practices.

<img width="539" alt="image" src="https://user-images.githubusercontent.com/51073515/114905039-7f607980-9de6-11eb-97c8-b547ae677c64.png">
`Next` button doesn't change color and is not clickable after login.

I wonder if we can fix this by changing how useAuth() passes values.

Reply:
OK, let's leave it as is, and change your default value in 139 from `false` to `!!user`, to properly reflect the initial value.

Reply:
Ok

Reply:
This problem persists after commit `Change request (#22)`

Reply:
Done.